pkg/k8s/pipelineRunFetcher.go

```diff
@@ -0,0 +1,87 @@
+package k8s
+
+import (
+	"fmt"
+
+	api "github.com/SAP/stewardci-core/pkg/apis/steward/v1alpha1"
+	stewardLister "github.com/SAP/stewardci-core/pkg/client/listers/steward/v1alpha1"
+	"github.com/pkg/errors"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+// PipelineRunFetcher has methods to fetch PipelineRun objects from Kubernetes
+type PipelineRunFetcher interface {
+	pipelineRunNameFetcher
+	ByKey(key string) (*api.PipelineRun, error)
+}
+
+type pipelineRunNameFetcher interface {
+	ByName(namespace, name string) (*api.PipelineRun, error)
+}
+
+type pipelineRunListerFetcher struct {
+	lister stewardLister.PipelineRunLister
+}
+
+// NewPipelineRunListerFetcher returns an operative implementation of PipelineRunFetcher
+func NewPipelineRunListerFetcher(lister stewardLister.PipelineRunLister) PipelineRunFetcher {
+	return &pipelineRunListerFetcher{
+		lister: lister,
+	}
+}
+
+// ByName fetches PipelineRun resource from Kubernetes by name and namespace
+// Return nil,nil if specified pipeline does not exist
```

Comment:
Put the docs on the interface to make them visible there. Here you just need to refer to the interface.